File: Backend/manga-world/src/main/java/com/amakakeru/mangaworld/Entity/Authorities.java
package com.amakakeru.mangaworld.Entity;

import jakarta.persistence.*;
import lombok.Data;

@Data
@Entity
@Table(name = "authorities")
public class Authorities {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "auth_id")
    private Long authId;

    @Column(name = "type", nullable = false)
    private String type;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false, updatable = false)
    private User user;
}

File: Frontend/manga-world/src/Model/Authorities.ts
import User from "./User";

class Authorities {
  authId: number;
  type: string;
  user: User;

  constructor(authId: number, type: string, user: User) {
    this.authId = authId;
    this.type = type;
    this.user = user;
  }
}

export default Authorities;
